src/logic/small.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import data.equiv.basic
+
+/-!
+# Small types
+
+A type is `w`-small if there exists an equivalence to some `S : Type w`.
+
+We provide a noncomputable model `shrink α : Type w`, and `equiv_shrink α : α ≃ shrink α`.
+
+A subsingleton type is `w`-small for any `w`.
```

Comment:
I think it would be nice to generalise this to say any fintype is small, though probably that can be in a future PR.

Reply:
It's easy enough. I've added it. I also added the instance direct from `subsingleton` to `fintype`, which apparently didn't exist previously because it's classical, and `fintype` was written back when people cared about this. :-)